codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -352,7 +355,9 @@ class HttpBindingGenerator(
             rustType to targetShape
         }
         val parsedValue = safeName()
-        if (coreType == dateTime) {
+        // We only check for the type name to allow other implementations that are swapping symbols like
+        // Python to execute the right branch and not fall back to the string deserialization.
+        if (coreType.name == dateTime.name) {
```

Comment:
I am not sure around relaxing this check. The main issue here is that since we are swapping out the standard `DateTime` symbol with the Python one, we don't end up in the right branch and we go in the else.

An option would be to check if the `AfterDeserializingIntoADateTimeOfHttpHeaders` customization is present to make the check stronger, something like in pseudocode:

```
if (AfterDeserializingIntoADateTimeOfHttpHeaders in customizations and coreType.name == dateTime.name) or (coreType == dateType) {
    ...
}
```

I really need some advices here.

Reply:
Matteo and I discussed offline—we'll change the check to look at the Smithy shape instead of the Rust type